File: content-solutions/implementation/sim-ladder.java
import java.util.*;

class Solution {
    public int[] solution(int n, int[][] bars) {
        int[][] sorted = bars.clone();
        Arrays.sort(sorted, (x, y) -> x[0] != y[0] ? Integer.compare(x[0], y[0]) : Integer.compare(x[1], y[1]));
        int[] at = new int[n];
        for (int i = 0; i < n; i++) at[i] = i;
        for (int[] bar : sorted) {
            int col = bar[1];
            int t = at[col];
            at[col] = at[col + 1];
            at[col + 1] = t;
        }
        int[] answer = new int[n];
        for (int col = 0; col < n; col++) answer[at[col]] = col;
        return answer;
    }
}

File: content-solutions/implementation/sim-ladder.js
function solution(n, bars) {
  const at = Array.from({ length: n }, (_, i) => i);
  for (const [, col] of [...bars].sort((a, b) => a[0] - b[0] || a[1] - b[1])) {
    [at[col], at[col + 1]] = [at[col + 1], at[col]];
  }
  const answer = new Array(n).fill(0);
  for (let col = 0; col < n; col++) answer[at[col]] = col;
  return answer;
}
